v6-dev

mdo opened the pull request:
Yes, it's happening! Here's what's happened so far:

## High level
- [x] Dropped extraneous grid, reboot, and utilities bundles for now (potentially will bring them back)
- [x] Browser support has been significantly updated, moving to floating targets (e.g., `last 1 versions` and `not dead`)
- [x] Moved from Hugo to Astro for the docs
- [x] Fixes #29919: Added Markdownlint via npm script (`npm run lint-mdx`)
- [ ] Monorepo structure?

## Sass & CSS
- [x] Fix #29853: Migrated to Sass modules across the board
- [x] All Dart Sass, no Node Sass
- [x] New `_colors.scss` stylesheet
- [x] New `_config.scss` stylesheet
- [x] Fix #28553: Rearranged Sass files (e.g., `_forms.scss` is now `forms/index.scss` and uses `@forward`)
- [x] Implement CSS layers
- [ ] Fix #40412: Drop custom `add()` and `subtract()` functions for native `calc()`; also replace custom `divide()` function with Sass `math.div()`.
- [x] Drop `!important` from utilities
- [ ] Native CSS nesting in output?
- [ ] Replace RFS with something simpler, maybe using `clamp()`?
- [ ] Fix #33263: Localize variables, mixins, and some functions within each stylesheet that needs them (except truly shared or global ones)
- [x] Fix #41046: Range CSS media queries (e.g. `@media (width < 768px)`

## Colors, Customizing, Themes, etc
- [x] Fix #39587: Add all color tints and shades as CSS variables
  - [ ] Migrate components to use these CSS variables
- [ ] Simplify theme support down to light/dark mode, media queries only instead of `data-bs-theme`?
- [ ] Use more `light-dark()` function?
- [ ] Ensure `color-scheme` values are being set
- [ ] If using `data-bs-theme` still, look into how we could set color/bg automatically?
- [ ] Move to oklch over mix of hex/rgb?
- [ ] Use native color-mix() as needed
- [ ] Fix #38569: Normalize gray color keys in `$grays` map to the other colors
- [ ] Use relative colors as needed (component variants and states?)
- [ ] P3 colors? Mixin to convert them? Plugin to do so after the fact?
- [ ] Expand the color palette more if we can? Move from semantic colors to full palette for component variants?

## Layout
- [x] Dropped OG flexbox grid and replaced with CSS Grid layout (includes dropping most mixins I think?)
- [x] Renamed `$grid-breakpoints` to `$breakpoints`
- [x] Renamed `-xxl` breakpoint to `-2xl`, as well as all component size variants
- [ ] Sub grid modifier to inherit parent grid cols
- [ ] Add autofill grid option plus 12-col system
- [ ] Explore additional flex utilities to support the move
- [ ] Expand width utilities perhaps? Since grid layout uses grid-specific properties

## Reboot
- [ ] Ditch legend and fieldset changes, add a fieldset reset perhaps
- [ ] Fix #40582: Colorize `code` elements for light/dark mode

## Content
- [x] Fix #30177: Add new `.prose` class for long form content

## Buttons
- [ ] Move to inline-flex
- [ ] Simplify btn-check code? Restyle checkboxes as button?

## Forms
- [ ] Replace custom radio SVG with background/border combo
- [ ] Improve input groups, especially with validation
- [ ] `:user-valid/invalid` vs `:valid/invalid`
- [ ] `field-sizing: content` for textareas?

## Utilities & Helpers
- [x] Fix #36129: Moved ratio helper to become `.ratio` utility
- Sidebar: You can now modify the utilities map to make these responsive. Original request at #39449.
- [x] Fix #38641: Drop clearfix helper for new `display: flow-root` utility, `.d-flow-root`
- [ ] Explore `@property` in CSS for disabling inheritance? (see appendix)
- [ ] Smooth out API?
- [ ] Fix #40674: Rename `max-width` and `max-height` utilities for clarity
- [ ] Fix #41330: Add new `min-content`, `max-content`, and `fix-content` values for width and height utilities

## JavaScript
- [x] Drop all things jQuery from our plugins
- [ ] Modernize modal plugin to become a new Dialog plugin, built on the native dialog component
  - [ ] Revisit #40787: add CSS variables for modal widths across breakpoints
- [ ] Popover native plugin? Dialog for popover to natively disable scroll? Anchor positioning (not yet in FF or Safari)
- [ ] @starting-style for animations?
- [ ] Native drop down menu?
- [ ] Add support to dropdowns for context menu
- [ ] New plugin ideas
  - [ ] PIN/2FA input
  - [ ] Toggle attribute
  - [ ] Date picker
  - [ ] Password strength
- [ ] Fixes #29378: Automatically make tooltips and popovers work, remove performance mention from docs

## Docs
- [x] Fix #39041: Add previous/next links at bottom of docs pages
- [ ] Add keyboard section to JS components to list supported keyboard interactions

---

## Appendix

Custom properties to disable inheritance in some utility classes?
```
@property --bs-bg-alpha {
  syntax: "<number>";
  inherits: false;
  initial-value: 1;
}
@property --bs-text-alpha {
  syntax: "<number>";
  inherits: false;
  initial-value: 1;
}
```

### Screenshots

Expansive color palette?

![CleanShot 2025-01-15 at 21 25 02@2x](https://github.com/user-attachments/assets/dbcd2a76-e447-4006-8b7b-2db8d584a45b)

More button variants, all generated from a larger map?

![CleanShot 2025-01-15 at 21 25 10@2x](https://github.com/user-attachments/assets/e6ba2d81-5249-4fa7-add7-01cd8de0860e)

CSS grid in action

![CleanShot 2025-01-15 at 21 29 21@2x](https://github.com/user-attachments/assets/caae0f53-0bca-4cd9-af50-c62e3c2b7b9d)

mdo committed:
Remove grid, reboot, and utilities bundles

mdo committed:
Migrate to Sass modules

fix function

mdo committed:
Remove css-prefix-* scripts other than main build

mdo committed:
Give helpers folder an index.scss, migrate ratio helper to aspect-ratio utility

mdo committed:
Remove some enable vars

mdo committed:
Generate classes for color palette instead of manually doing it

mdo committed:
Implement CSS layers

Rearrange and comment import stack

Move content stylesheets to new folder

mdo committed:
Migrate from flex to CSS grid layout

ffoodd commented:
Wow, what a nice move 🚀

Any way to share ideas for v6? Or should we try some PR to your branch?

julien-deramond commented:
Adding this link as a reference for reusing other elements in case we run into any issues: https://github.com/twbs/bootstrap/pull/38319

mdo commented:
@ffoodd Ayyy good to see you! Issues here or jump into our team Slack :). I'll update the original comment with more of what I'm thinking shortly.

mdo committed:
It's over 9000

mdo committed:
Breakpoints edits

Breakpoints update

- Rename $grid-breakpoints to $breakpoints
- Rename -xxl to -2xl
- Update docs to use new classes
- Update breakpoints docs to add new language around updated breakpoint mixins and media queries, including no longer needing -.02px

mdo committed:
All utilities are no longer !important; they use the top CSS layer instead so they're always applied

mdo committed:
Remove jQuery support in plugins

mdo committed:
Drop clearfix helper for display: flow-root

mdo committed:
Remove add(), subtract(), and divide() functions for calc() and Sass math.div()

ffoodd commented:
💜

I only have a few things that comes to my mind : 
- switch to logical properties, to drop RTLCSS and RTL build ;
- try to rely on modern HTML for components : `<dialog>`, `<details>` and `<summary>`, maybe `popover` and its API ?
- use container queries for responsive ?

Probably for v7 but the anchor
positioning API could replace Popper / Floating UI, but we may try as a progressive enhancement. 

There could be more, but those are already quite big.

mdo committed:
Fix stylelint

mdo committed:
Update Alerts variables and remove deprecation mentions

mdo committed:
Add Markdownlint for our MDX

mdo committed:
WIP on new .content-body class

mdo committed:
Browserslist: move into package.json, make super agressive

mdo committed:
New _colors.scss

mdo committed:
Reorganize scss folder

mdo committed:
Migrate docs to Sass modules, comment out docs grid CSS

mdo committed:
Use CSS variable for border-radius in utilities

mdo committed:
More docs Sass modules

mdo committed:
fix some docs Sass

mdo committed:
WIP, needs to be broken apart

- button ideas
- rewrite utility mixin
- cleanup some code
- remove hyphen from infix by default

mdo committed:
New .prose component, put it to work in the docs

mdo committed:
Revamp some docs layout

mdo committed:
Drop contents and best practices docs pages

mdo committed:
Revamp theme colors and more

mdo committed:
More docs stuff and wip edits

mdo committed:
WIP scss

mdo committed:
remove rtl stuff

mdo committed:
Remove RTL package

mdo committed:
Toggler compontent from upstrea PR

mdo committed:
Rename mh-* and mw-* to max-h/w-*, add additional width and height values

Fixes #41330, fixes #40674.

mdo committed:
Fix deps and watch script

mdo committed:
Don't import Bootstrap CSS again

mdo committed:
Move around box shadow vars

mdo committed:
Tweak docs layout and font-sizes, try out Inter

mdo committed:
Delete _maps.scss

mdo committed:
New grid gap, tweak form layouts in docs

mdo committed:
New focus styles

mdo committed:
Fix RFS in utilities

mdo commented on .github/workflows/browserstack.yml:
Testing

mdo committed:
Fix masthead, remove Inter

mdo committed:
Redo background and color utilities with new classes and more

mdo committed:
List spacing improvement

mdo committed:
Fix stylelint problems

mdo committed:
Mirror docs content change for nested list vertical margin

mdo committed:
Redo aspect ratio docs

mdo committed:
split border-radius, new utility reference table, start adding mdn link

mdo committed:
Move ads to right sidebar to reduce reflow

mdo committed:
fixes

mdo committed:
CSS Tricks links too

mdo committed:
Edits

mdo committed:
tweaks

mdo committed:
fix some css vars

mdo committed:
Linter fix

mdo committed:
Fix some utils

mdo committed:
utility links

mdo committed:
widen gaps in docs

mdo committed:
Migrate some Sass vars to be local to specific stylesheets

- Update ScssDocs tags where appropriate
- Removed a null Sass variable, $table-th-font-weight
- Removed $table-caption-color because we put table Sass vars in the _tables.scss file away from the _reboot.scss file
- Fix docs styles as needed

mdo committed:
Modify Sass tests to pass for new Sass modules implementation

mdo committed:
Remove Node Sass workflow for Actions

mdo committed:
Update bundlewatch to remove deleted bundles, modify CSS maxSize values

mdo committed:
Fix some spelling issues, replace lorem ipsum with generated prose

mdo committed:
Fix malformed HTML in modal

mdo committed:
Another typo

mdo committed:
Remove jQuery refs from JS unit test

mdo committed:
Fix an issue in utilities map, update scssdocs across docs files, remove migration for now

mdo committed:
Fix CSS vares fileMatch

mdo committed:
Fix JS linter errors

mdo committed:
Remove broken links

mdo committed:
fix linter issues
